File: LeetCode_848/Main.java


// class Solution {
//     public String shiftingLetters(String s, int[] shifts) {
//         long op = 0;
//         char[] chars = s.toCharArray();
//         for (int i = shifts.length - 1; i >= 0; i--) {
//             op += shifts[i];
//             int d = chars[i] - 'a';
//             d = (int)((d + op) % 26);
//             chars[i] = (char)(d + 'a');
//         }
//         return new String(chars);
//     }
// }




public class Main {

    public static void main(String[] args) {
        
    }
}

File: LeetCode_848/Main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char* shiftingLetters(char* s, int* shifts, int shiftsSize) {
    long long op = 0;
    int len = strlen(s);
    for (int i = shiftsSize - 1; i >= 0; i--) {
        op += shifts[i];
        int d = s[i] - 'a';
        d = (d + op) % 26;
        s[i] = d + 'a';
    }
    return s;
}
